Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -2951,9 +3022,13 @@ PhysicalParticleContainer::getIonizationFunc (const WarpXParIter& pti,
                                 ion_atomic_number};
 }
 
-PlasmaInjector* PhysicalParticleContainer::GetPlasmaInjector ()
+PlasmaInjector* PhysicalParticleContainer::GetPlasmaInjector (int i)
 {
-    return plasma_injector.get();
+    if (i < 0 || i >= static_cast<int>(plasma_injectors.size())) {
+        return nullptr;
```

Comment:
Should we call `amrex::Abort` or `WARPX_ALWAYS_ASSERT_WITH_MESSAGE` here?

Reply:
This routine is now only used by the moving window code, and it checks if the result is a null pointer. There can be species for which the injection was not defined in which case `plasma_injectors.size()` will be zero, so the call from the moving window passing in zero would then get the null pointer.